Issue/revisions user details

khaykov opened the pull request:
This PR adds functionality for retrieving details of revision authors and injecting them into the revision list.

After we done retrieving revisions, we gather ID of revision authors, and use them to retirieve their details by utilizing `/batch/`  and `/sites/$site/users` endpoing with search parameter.

Utilizing `/batch/` endpoint is a bit hacky, but it works. 

To test:
1. Open Revision for a post with multiple authors.
2. Make sure their avatars are displayed properly.

PS: This PR does not incliudes the Display Names of Revision authors. @iamthomasbishop If you would like to add it to the design let me know :)

[![Image from Gyazo](https://i.gyazo.com/6f804cbcdc597894288a7d2f121ce7fb.png)](https://gyazo.com/6f804cbcdc597894288a7d2f121ce7fb)

khaykov commented:
No idea why Travis is grumpy about the build...

theck13 commented on WordPress/src/main/java/org/wordpress/android/viewmodel/history/HistoryViewModel.kt:
```diff
@@ -98,7 +115,40 @@ class HistoryViewModel @Inject constructor(
+            }
+
+            override fun onError() {
+                AppLog.e(AppLog.T.POSTS, "Can't fetch details of post revision authors ")
```
```suggestion
                AppLog.e(T.API, "Can't fetch details of revision authors")
```

We can remove the `AppLog` qualification from `AppLog.T.POSTS` since `org.wordpress.android.util.AppLog.T` is an import statement.

Since `HistoryViewModel` is used for both pages and posts, we should remove the `post` part of the error message and update the `POSTS` log type to `API`.

khaykov commented:
Thanks for the review, Tyler!
I fixed the issues and added the display name as we discussed.
[![Image from Gyazo](https://i.gyazo.com/9342a83bd028782a747d0c42014bb1a9.png)](https://gyazo.com/9342a83bd028782a747d0c42014bb1a9)

iamthomasbishop commented:
This is looking really solid! Are we only showing display name on multi-author sites?

theck13 commented:
The display name is shown for multi-author and single-author sites.  Since we're showing the avatar in both cases, showing the display name make sense too.  If we didn't want to show the display name for single-author sites, what should we display for the avatar and subtitle?

khaykov commented:
Cool! I update the title to use `formattedTime`, now it looks like this:
[![Image from Gyazo](https://i.gyazo.com/0f97ccbedb9390603317bd6026b6ee27.png)](https://gyazo.com/0f97ccbedb9390603317bd6026b6ee27)
